exporter/instanaexporter/exporter.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instanaexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter"
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"runtime"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter/internal/backend"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter/internal/converter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter/internal/converter/model"
+)
+
+type instanaExporter struct {
+	config    *Config
+	client    *http.Client
+	settings  component.TelemetrySettings
+	userAgent string
+}
+
+func (e *instanaExporter) start(_ context.Context, host component.Host) error {
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		return err
+	}
+	e.client = client
+	return nil
+}
+
+func (e *instanaExporter) pushConvertedTraces(ctx context.Context, td ptrace.Traces) error {
+	converter := converter.NewConvertAllConverter(e.settings.Logger)
+	spans := make([]model.Span, 0)
+
+	hostID := ""
+	resourceSpans := td.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resSpan := resourceSpans.At(i)
+
+		resource := resSpan.Resource()
+
+		hostIDAttr, ex := resource.Attributes().Get(backend.AttributeInstanaHostID)
+		if ex {
+			hostID = hostIDAttr.StringVal()
+		}
+
+		ilSpans := resSpan.ScopeSpans()
+		for j := 0; j < ilSpans.Len(); j++ {
+			converterBundle := converter.ConvertSpans(resource.Attributes(), ilSpans.At(j).Spans())
+
+			spans = append(spans, converterBundle.Spans...)
+		}
+	}
+
+	bundle := model.Bundle{Spans: spans}
+
+	if len(bundle.Spans) == 0 {
```

Comment:
Would this commonly happen? I would expect a warning message to exist somewhere if spans are being received but not exported.

Reply:
Probably not but is it better incase such a scenario arises we are unaware of now?

Reply:
What do you think?

Reply:
I guess there are two situations: your component received an empty batch, in which case, it's OK to have an empty outcome. I'm not sure we _should_ catch this in the component, though. 

Then, there's the situation where you get a non-empty batch, but you filter out spans and end up with an empty batch. In that case, it would be good to have a metric stating how many spans were received and how many were filtered out. Users can then use those numbers to debug and understand what's going on.

It doesn't look like you are filtering anything, though, so I can't see whether this case will ever happen. If you are indeed not filtering anything, just get rid of this condition.

Reply:
If you prefer to deal with this in a subsequent PR, that's fine by me.

Reply:
Ok, will think about it and deal with it in another PR.